app.rb

```diff
@@ -104,4 +151,77 @@ def rental_list
       end
     end
   end
+
+  private
+
+  def load_books
+    return if @books_data.empty?
+
+    books_array = JSON.parse(@books_data)
+    books_array.each do |book|
+      @books << Book.new(book['Title'], book['Author'])
+    end
+  end
+
+  def load_people
+    return if @people_data.empty?
+
+    people_array = JSON.parse(@people_data)
+    people_array.each do |person|
+      @people << if person['type'] == 'Student'
+                   Student.new(person['age'], nil, parent_permission: person['parent_permission'],
+                                                   name: person['name'])
+                 else
+                   Teacher.new(person['age'], person['specialization'],
+                               parent_permission: person['parent_permission'], name: person['name'])
+                 end
+    end
+  end
+
+  def process_rental(rental)
+    selecting_book = @books.find { |book| book.title == rental['Book'] }
+    selecting_people = @people.find { |person| person.name == rental['Person'] }
+
+    if selecting_book.nil?
+      puts "Book not found: #{rental['Book']}"
+      return
+    end
+
+    if selecting_people.nil?
+      puts "Person not found: #{rental['Person']}"
+      return
+    end
+
+    @rentals << Rental.new(rental['Date'], selecting_book, selecting_people)
+  end
+
+  def save_books
+    book_json = @books.map do |book|
+      { Title: book.title, Author: book.author }
+    end
+
+    File.write('./data/book.json', JSON.generate(book_json))
+  end
+
+  def save_people
+    people_json = @people.map do |person|
+      if person.type == 'Student'
+        people_json.push({ type: person.type, name: person.name, age: person.age,
+                           parent_permission: person.parent_permission })
+      else
+        people_json.push({ type: person.type, name: person.name, age: person.age,
+                           parent_permission: person.parent_permission, specialization: person.specialization })
+      end
+    end
```

Comment:
- [ ] After creating book, people, and rentals, it's expected that when option 7 is clicked, the app closes successfully and persists the entered data into JSON files, but unfortunately, the app crashes with the error shown in the screenshot below. Kindly work on this so data is saved successfully and is fetched upon reload of the app. 👍🏽

<img width="1213" alt="Screenshot 2023-07-14 at 8 50 00 AM" src="https://github.com/BranBayou/oop-class-library/assets/50941074/5be0a25d-05d3-4320-a717-b6e7431b421d">